### PR TITLE
feat: v2 history endpoint with permissions

### PR DESCRIPTION
**Pull request by tim-schultz**
fixes: https://github.com/passportxyz/passport/issues/2956

### CODE REVIEW

File: api/registry/api/common.py [hunk 1]
```diff
@@ -246,6 +253,9 @@ def get_score_history(
 If a specific address is provided, then the list will be of max length 1.\n
 \n
 Note: results will be sorted descending by `["created_at", "id"]`
+\n
```
Comment (nutrina) [inline api/registry/api/common.py:256]: do we need this `history_endpoint` construct?
Why not just specify all this information in the api handler decorator itself

Comment (tim-schultz) [reply api/registry/api/common.py:99]: I was thinking we would need it for specifying permissions specifically for this endpoint. It looks like this is how the rest of the api key permissions are being used

Comment (nutrina) [reply api/registry/api/common.py:99]: Would be able to handle that in the ApiKey class? That is then used from both the Ninja API and the lambdas.

Comment (tim-schultz) [reply api/registry/api/common.py:99]: ✅

**Pull request merged**